File: packages/types/src/insight.ts
export interface Insight {
  id: string
  projectId: string
  title: string
  content: string
  sources: Array<{ sourceId: string; sourceTitle: string }>
  createdAt: string
}

export interface GenerateInsightRequest {
  title: string
  sourceIds: string[]
  prompt?: string
}

File: apps/api/app/schemas/insight.py
import uuid

from pydantic import BaseModel


class InsightGenerateRequest(BaseModel):
    title: str
    source_ids: list[str]
    prompt: str | None = None


class InsightResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    title: str
    content: str
    sources: list[dict]

    model_config = {"from_attributes": True}
